### PR TITLE
Worker: emit "died" before observer "close"

### PR DESCRIPTION
**Pull request by ibc**
Replaces PR #682

Tests added.

### CODE REVIEW

**Review by piranna — APPROVED** [review]
LGTM.

File: node/tests/test-Worker.js [hunk 1]
```diff
@@ -161,54 +161,109 @@ test('worker.close() succeeds', async () =>
...
+	onDied = jest.fn();
 	onObserverClose = jest.fn();
 
 	worker.observer.once('close', onObserverClose);
 
-	await new Promise((resolve) =>
+	await new Promise((resolve, reject) =>
 	{
-		worker.on('died', resolve);
+		onDied();
```
Comment (piranna) [inline node/tests/test-Worker.js:243]: This should be inside the `died` event handler.

Comment (ibc) [reply node/tests/test-Worker.js:243]: I'll check later

Comment (ibc) [reply node/tests/test-Worker.js:243]: Fixed in v3, thanks

Comment (piranna) [reply node/tests/test-Worker.js:243]: Welcome :-)

**Pull request merged**